### PR TITLE
Re-add missing resize event to news window

### LINKED ISSUE #2316: (Minor) News box is behaving strangely
**Version information:**
 - Platform: Windows 10
 - Version: 24.02.1
 - Build/commit: 8e40251e

**Describe the bug**
The pop-up news box has developed a couple of bugs. One is that a small strip of the box is not rendered, leaving a 'window' on the landscape behind.

Also, when new vehicles are announced, the 3D images spin erratically or not at all, and the image gets corrupted.

**To Reproduce**
1. Wait for a pop-up news box to appear announcing new vehicles.

**Expected behaviour**
Previous versions seemed to work ok.

**Screenshots**

spinner

### PR DESCRIPTION
**Pull request by AaronVanGeffen**
Regression from #2304. Should fix #2316, #2321.

### CODE REVIEW

File: src/OpenLoco/src/Windows/News/News.cpp [hunk 1]
```diff
@@ -851,6 +851,7 @@ namespace OpenLoco::Ui::Windows::NewsWindow
 
         static constexpr WindowEventList kEvents = {
             .onMouseUp = onMouseUp,
+            .onResize = initViewport,
```
Comment (ZehMatt) [inline src/OpenLoco/src/Windows/News/News.cpp:854]: I would somewhat prefer if we have an explicit functions for onResize and viewportRotate, this feels prone to accidentally do something it shouldn't.

Comment (AaronVanGeffen) [reply src/OpenLoco/src/Windows/News/News.cpp:854]: I would prefer that, too. However, looking at `initViewport` in its current state, I believe it's overdue a refactor, anyway. I would prefer for it to be split up into several functions, and made into different events then.

Comment (ZehMatt) [reply src/OpenLoco/src/Windows/News/News.cpp:854]: I see, better left for another PR then.

**Review by ZehMatt — APPROVED** [review]

**Review by LeftofZen — APPROVED** [review]
tested, had no issues with rotating vehicles in the annoucement video. looks like it works!

**Pull request merged**